eq-author-api/middleware/runQuestionnaireMigrations.test.js

```diff
@@ -0,0 +1,77 @@
+const runQuestionnaireMigrations = require("./runQuestionnaireMigrations");
+const { buildQuestionnaire } = require("../tests/utils/questionnaireBuilder");
+
+describe("runQuestionnaireMigrations", () => {
+  let res, req, next, migrations, logger;
+
+  beforeEach(async () => {
+    req = {
+      questionnaire: await buildQuestionnaire({
+        metadata: [{}],
+      }),
+    };
+    res = {};
+    next = jest.fn();
+    migrations = [];
+    logger = {
+      info: jest.fn(),
+    };
+  });
+
+  it("should correctly handle no questionnaire", async () => {
+    req = {};
+    let result = await runQuestionnaireMigrations(logger)(migrations)(
+      req,
+      res,
+      next
+    );
+    expect(next).toHaveBeenCalled();
+    expect(result).toBeFalsy();
+  });
+
+  it("should correctly handle an undefined version", async () => {
+    req.questionnaire.version = undefined;
+    let migrationOne = jest.fn(() => req.questionnaire);
+    migrations = [migrationOne];
+
+    await runQuestionnaireMigrations(logger)(migrations)(req, res, next);
+
+    expect(next).toHaveBeenCalled();
+    expect(migrationOne).toHaveBeenCalledWith(req.questionnaire);
+    expect(req.questionnaire.version).toEqual(1);
+  });
+
+  it("should not migrate data if version up to date", async () => {
+    req = {
+      questionnaire: {
+        version: 1,
+      },
+    };
+    migrations = [jest.fn()];
+    let result = await runQuestionnaireMigrations(logger)(migrations)(
+      req,
+      res,
+      next
+    );
+    expect(next).toHaveBeenCalled();
+    expect(result).toBeFalsy();
```

Comment:
Can we assert on no migrations being run instead of result which does not require this functionality for it to be falsy?